### ISSUE TITLE
Monitoring not working correctly with Histograms that have fit functions

### ISSUE THREAD

**Issue by daritter** [root-project/jsroot#125]
I have a very minimal histfitserver.C which only fills one histogram with gaussian values and fits this histogram periodically:

```
void histfitserver(void)
{
  auto serv = new THttpServer("http:8081");
  auto h1 = new TH1F("h1", "histogram 1", 100, -5, 5);
  auto c1 = new TCanvas("c1");
  auto f1 = new TF1("f1", "gaus", -10, 10);

  while (true) {
    h1->FillRandom("gaus", 100);
    c1->cd();
    h1->Draw();
    h1->Fit(f1);
    c1->Modified();
    c1->Update();
    gSystem->ProcessEvents();
    sleep(1);
  }
}
```

Now opening this in the browser with http://localhost:8081?monitoring=1000 I can see the histogram and the canvas and I can open them and it also draws the fit function as expected. 

The contents of the histogram are updated as expected but the fit function is never updated: 
![hist-with-function](https://cloud.githubusercontent.com/assets/1186338/24150906/7a07daa6-0e47-11e7-957d-9d5fc05d29d2.png)
Upon reload the function is redrawn at the correct position but not updated. I checked the transferred data and the correct function parameters seem to be transferred in the json object.

It does work as expected if I don't add the TF1 to the list of functions but draw it separately `h1->Fit(f1, "N"); f1->Draw("same");`

**Comment by linev** [comment]
I can reproduce problem.
Hope to fix it soon.

**Comment by linev** [comment]
Hi, it was simple bug - now fixed.
To use fixed version, you can checkout jsroot repository and set JSROOTSYS shell variable like

    [shell] export JSROOTSYS=/home/user/jsroot

In that case THttpServer will use alternative location for JSROOT.

I plan bugfix release this week, then code will be submitted to ROOT repository.

**Comment by b-spruck** [comment]
Thank you. Maybe the above script could be added as example code. E.g. that you need the Modified() line might help other people not to waste their time.

**Comment by linev** [comment]
Ok, I will add macro in tutorials after next JSROOT release.

**Comment by linev** [comment]
Actually, h1->Draw() normally should be called once.
You can see how it done in tutorials/hsimple.C

**Comment by b-spruck** [comment]
hsimple.C fills it every 1000 events. at least in my version.
But yes, I tried to Draw only once and make only the fit and Modified/Update in "event" loop.

**Comment by linev** [comment]
Hi
Now fix is in [5.1.1](https://github.com/root-project/jsroot/releases/tag/5.1.1) release and in current ROOT master. I also add your script (with small modifications) in http tutorials:

https://github.com/root-project/root/blob/master/tutorials/http/histfitserver.C

Regards,
Sergey
